### ISSUE TITLE
Open to run scripts with silent flag

### ISSUE THREAD

**Issue by marklawlor** [vercel/turborepo#672]
### Current output:

```
my-project:test:
my-project:test: > my-project@0.0.0 test
my-project:test: > echo 'hello-world'
my-project:test:
my-project:test: hello world
```

### Output I'd like

```
my-project:test: hello world
```

### Describe the solution you'd like

`npm run` has a `--silent` option that removes the extra lines added by `npm` when running a script. I would like an option in `turbo.json` to pass this flag to `npm run`.

### Describe alternatives you've considered

Have an environment variable `NPM_OPTIONS` to pass extra command line flags to `npm` (similar to `NODE_OPTIONS` for node).

**Comment by andyjy** [comment]
Appears to achieve the desired effect by passing `--silent` after a bare double-dash
_(thus passing the `--silent` flag through to the npm run command rather than interpreting as a flag for `turbo`)_ -

`turbo run test -- --silent`

Agree it would be pleasing to have this feature natively in Turborepo config to avoid the need for shell jitsu.

**Comment by andyjy** [comment]
Turns out `npm` supports passing config via environment variables using the `npm_config_` prefix, so e.g. this does the trick:

`npm_config_loglevel=silent turbo run test`

https://docs.npmjs.com/cli/v8/using-npm/config#environment-variables
https://docs.npmjs.com/cli/v8/using-npm/config#loglevel

**Comment by leondreamed** [comment]
Has anybody found a solution that works for `pnpm`?

**Comment by PacoSoftwerft** [comment]
As I sometimes run turbo via npm it may be helpful to know you can chain the double dash.
Example:

```shell
npm run xy -- -- --flag
```

What happens `npm run xy -- -- --flag` -> `turbo run xy -- --flag` -> `xy --flag`

**Comment by net** [comment]
For pnpm add

```
reporter=silent
```

to your `.npmrc` (https://pnpm.io/npmrc).

This setting is undocumented, but it's equivalent to `--silent`: https://github.com/pnpm/pnpm/blob/main/pnpm/src/shorthands.ts#L8

**Comment by anthonyshew** [comment]
Hey, folks. I'm going to make the decision here to not add this feature for now.

My reasoning is that the package managers have these configurations themselves:

| Package Manager | Config Mechanism | How
|-----|-----|-----|
| npm | `.npmrc` | `loglevel=silent`
| pnpm | `.npmrc` | `reporter=silent`
| Yarn Classic | `.yarnrc` | `--*.silent true` in `.yarnrc`
| Yarn Berry | N/A | Silent by default
| Bun | N/A | Silent by default

Building a great product is making good decisions about what to add - and what not to add. Hopefully this is one of the latter. 😄 If something changes in the future, we can revisit this, but its not looking like this is making sense for today.

**Comment by net** [comment]
@anthonyshew did you see what I said about how `reporter=silent` is not an ideal solution to this issue for pnpm?

> Unfortunately this silences all pnpm commands (including pnpm i).

Please reconsider.

**Comment by marcesengel** [comment]
@anthonyshew @net yep, to reiterate: it mutes output for stuff like `pnpm install`, `pnpm add`, `pnpm ls` etc., which is far from optimal. Missing progress indicators, warnings, prompts to run `pnpm approve-builds` etc.

We need to be able to configure this in `turbo`, there is no proper builtin solution for either `npm` nor `pnpm`.
I think `yarn` supports per-command log level config.

**Comment by anthonyshew** [comment]
This is an ask to make in your JS package manager of choice, not Turborepo. The behavior being asked for is a request for different behavior at a different layer than Turborepo.
